experimental/examples/bubblebench/sharedtree/src/proxy/tree.ts

```diff
@@ -138,10 +139,10 @@ export class TreeArrayProxy<T> implements IArrayish<T> {
         const removedId = itemIds[itemIds.length - 1];
         const removed = getChild(this.tree, removedId, this.update);
         this.update(Delete.create(StableRange.only(this.tree.currentView.getChangeNode(removedId))));
-        return removed as T;
+        return removed as Serializable<T>;
     }
 
-    push(...item: T[]): number {
+    push(...item: Serializable<T>[]): number {
```

Comment:
this is a case i've struggled with a bit. ideally we could prevent creation of the type if T isn't serializable. i wonder if a static create would work, `create<C>(..):TreeArrayProxy<Serializable<C>>`